vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/AbstractGridSingleSelectionModel.java

```diff
@@ -72,9 +72,18 @@ public void select(T item) {
             return;
         }
         T oldItem = selectedItem;
-        doSelect(item, false);
 
-        getGrid().doClientSideSelection(Collections.singleton(item));
+        // Check if it is in case setItems
```

Comment:
This just mentions one use-case which causes an issue but it's doesn't say anything about why the following code is needed in case `setItems` is used...

Reply:
changed to fix the bug in gridConnector.js